applications/asset_tracker_v2/src/main.c

```diff
@@ -0,0 +1,481 @@
+ * in the event manager handler, and then queued up in the message queue
+ * for processing in the main thread.
+ */
+struct app_msg_data {
+	union {
+		struct cloud_module_event cloud;
+		struct ui_module_event ui;
+		struct sensor_module_event sensor;
+		struct data_module_event data;
+		struct util_module_event util;
+		struct modem_module_event modem;
+		struct app_module_event app;
+	} module;
+};
+
+/* Application module super states. */
+static enum state_type {
+	STATE_INIT,
+	STATE_RUNNING
+} state;
+
+/* Application sub states. The application can be in either active or passive
+ * mode.
+ *
+ * Active mode: Sensor data and GPS position is acquired at a configured
+ *		interval and sent to cloud.
+ *
+ * Passive mode: Sensor data and GPS position is acquired when movement is
+ *		 detected, or after the configured movement timeout occurs.
+ */
+static enum sub_state_type {
+	SUB_STATE_ACTIVE_MODE,
+	SUB_STATE_PASSIVE_MODE,
+} sub_state;
+
+/* Internal copy of the device configuration. */
+static struct cloud_data_cfg app_cfg;
+
+/* Timer callback used to signal when timeout has occurred both in active
+ * and passive mode.
+ */
+static void data_sample_timer_handler(struct k_timer *timer);
+
+/* Application module message queue. */
+#define APP_QUEUE_ENTRY_COUNT		10
+#define APP_QUEUE_BYTE_ALIGNMENT	4
+
+K_MSGQ_DEFINE(msgq_app, sizeof(struct app_msg_data), APP_QUEUE_ENTRY_COUNT,
+	      APP_QUEUE_BYTE_ALIGNMENT);
+
+/* Data sample timer used in active mode. */
+K_TIMER_DEFINE(data_sample_timer, data_sample_timer_handler, NULL);
+
+/* Movement timer used to detect movement timeouts in passive mode. */
+K_TIMER_DEFINE(movement_timeout_timer, data_sample_timer_handler, NULL);
+
+/* Movement resolution timer decides the period after movement that consecutive
+ * movements are ignored and do not cause data collection. This is used to
+ * lower power consumption by limiting how often GPS search is performed and
+ * data is sent on air.
+ */
+K_TIMER_DEFINE(movement_resolution_timer, NULL, NULL);
+
+/* Module data structure to hold information of the application module, which
+ * opens up for using convenience functions available for modules.
+ */
+static struct module_data self = {
+	.name = "app",
+	.msg_q = &msgq_app,
+};
+
+/* Convenience functions used in internal state handling. */
+static char *state2str(enum state_type new_state)
+{
+	switch (new_state) {
+	case STATE_INIT:
+		return "STATE_INIT";
+	case STATE_RUNNING:
+		return "STATE_RUNNING";
+	default:
+		return "Unknown";
+	}
+}
+
+static char *sub_state2str(enum sub_state_type new_state)
+{
+	switch (new_state) {
+	case SUB_STATE_ACTIVE_MODE:
+		return "SUB_STATE_ACTIVE_MODE";
+	case SUB_STATE_PASSIVE_MODE:
+		return "SUB_STATE_PASSIVE_MODE";
+	default:
+		return "Unknown";
+	}
+}
+
+static void state_set(enum state_type new_state)
+{
+	if (new_state == state) {
+		LOG_DBG("State: %s", log_strdup(state2str(state)));
```

Comment:
I don't think any of these need to be given to log_strdup() either. The sub_state2str() function is returning string literals, which should be allocated in ROM.

Reply:
You're right, they should be, so `log_strdup()` has been removed